sdk/core/core/test/inc/az_test_span.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_test_span.h
+ *
+ * @brief Test only utilities for helping validate spans.
+ */
+
+#ifndef _az_SPAN_TESTING_H
+#define _az_SPAN_TESTING_H
+
+#include <assert.h>
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <cmocka.h>
+
+#include <az_result.h>
+#include <az_span.h>
+
+#include <stdio.h>
```

Comment:
nit: move this include with the group above (with stdint.h, etc.). Not sure why this one in particular was separated out.

Also, do we need it?